pkg/sql/parser/scanner_test.go

```diff
@@ -355,6 +355,9 @@ func TestScanError(t *testing.T) {
 		{`$0`, "placeholder index must be between 1 and 65536"},
 		{`$9223372036854775809`, "placeholder index must be between 1 and 65536"},
 		{`B'123'`, `"2" is not a valid binary digit`},
+		{string([]byte{34, 103, 111, 108, 97, 110, 103, 0x0, 34}), `control character found`},
```

Comment:
can we add a few test cases that don't use raw bytes, so it's a bit more readable?

for example:
```
{string("cat\ndog"), `control character found`},
```

Reply:
The string has to include the actual double quotes (representing the IDENT) so it'll have to be something like
``"cat\ndog"`` actually not sure if the \n is still considered a control character inside the ````

Maybe``"cat` + "\n" + `dog"``

Reply:
then we could do either 

```
{"\"cat\ndog\"", `control character found`},
```
or 
{`"cat
dog"`), `control character found`},